spring-xd-spark-streaming/src/main/java/org/springframework/xd/spark/streaming/java/ModuleExecutor.java

```diff
@@ -72,7 +73,10 @@ public void call(Iterator<?> results) throws Exception {
 									}
 								}
 								while (results.hasNext()) {
-									messageSender.send(MessageBuilder.withPayload(results.next()).build());
+									Object next = results.next();
+									Message message = (next instanceof Message) ? (Message) next :
+											MessageBuilder.withPayload(next).build();
```

Comment:
This particular use of MessageBuilder occurs on the worker node on the Spark cluster so it would not be within the same context as XD itself (on the container side). That said, we should consider whether that instance from the context could be serialized. Post 1.1 of course. @ilayaperumalg can you create a JIRA for that?